lib/src/messaging_client.dart

```diff
@@ -0,0 +1,159 @@
+
+import 'package:flutter_aira/src/models/sent_file_info.dart';
+import 'package:flutter_aira/src/platform_client.dart';
+import 'package:flutter_aira/src/platform_exceptions.dart';
+import 'package:logging/logging.dart';
+import 'package:pubnub/pubnub.dart' as pn;
+
+import 'models/message.dart';
+import 'models/session.dart';
+
+/// Room independent class to handle all messaging.
+abstract class MessagingClient {
+  /// The ID of the service request.
+  set serviceRequestId(int id);
+
+  /// A broadcast stream of messages sent and received.
+  ///
+  /// If the application does not support messaging, this will throw an exception.
+  Stream<Message> get messageStream;
+
+  Future<void> sendStart();
+
+  /// Sends the provided message to the Agent.
+  ///
+  /// If the application does not support messaging, this will throw an exception.
+  Future<void> sendMessage(String text);
+
+  /// Sends the provided message to the Agent.
+  ///
+  /// If the application does not support messaging, this will throw an exception.
+  Future<SentFileInfo> sendFile(String fileName, List<int> file, {String? text});
+
+  Uri getFileUrl(String fileId, String fileName);
```

Comment:
No comments? This method feels like it's exposing internals. Couldn't we call this behind the scenes when publishing to the broadcast stream?